### PR TITLE
IIS scenario: stdout and stderr are not present

### PR DESCRIPTION
**Pull request by shhsu**

### CODE REVIEW

**Comment by jkotalik** [comment]
dotnet test this

**Comment by jkotalik** [comment]
@dotnet-bot test this please

**Comment by jkotalik** [comment]
@pan-wang when running on IIS, stdout and stderr are set to -1, which causes _dup to return -1. Calling _dup2 with -1 causes an exception rather than returning -1, so we now explicitly check for it being -1.

**Comment by pan-wang** [comment]
when WAS disable stdout and stderr when creating worker process. This is why I said dup2 does not work in the code comments and called another API.

**Comment by jkotalik** [comment]
We don't use dup2 to redirect stdout/err to the file/pipe, we use it to store/restore the previous stdout that was set by IIS. Different scenario, but you are correct that it doesn't work as expected.

**Comment by pan-wang** [comment]
with your new change, does the logging work?

**Comment by jkotalik** [comment]
@shhsu can you verify that locally and see if stdout logs are put in the windows event log if there is an exception?

**Comment by jkotalik** [comment]
Pending verification.

**Comment by jkotalik** [comment]
I can take over this PR. There are few more issues left over with calling dup2 that I didn't realize. @pan-wang was right about calling dup2 I believe. Will revert appropriately.

**Comment by jkotalik** [comment]
Going to revert to original behavior and disable tests for now.

**Comment by shhsu** [comment]
Just to confirm: this is just a manual revert?

**Comment by jkotalik** [comment]
Yeah, we still moved the location of output redirection to another file.

File: src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp [hunk 1]
```diff
@@ -147,38 +137,56 @@ FileOutputManager::Start()
...
     }
 
-    // Periodically flush the log content to file
-    m_Timer.InitializeTimer(STTIMER::TimerCallback, &m_struLogFilePath, FILE_FLUSH_TIMEOUT, FILE_FLUSH_TIMEOUT);
+    if (!SetStdHandle(STD_ERROR_HANDLE, m_hLogFileHandle))
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        return hr;
+    }
+    //int nHandle = _open_osfhandle((intptr_t)m_hLogFileHandle, 0);
```
Comment (pan-wang) [inline src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp]: please clean  those lines  (171-178) if you don't need them

**Comment by jkotalik** [comment]
@shhsu @pan-wang 🆙 📅

**Comment by shhsu** [comment]
LGTM

**Pull request merged**